### ISSUE TITLE
JEI Crashing due to Portable Stone Cutter

### ISSUE THREAD

**Issue by jbringrose** [brckts/Portable-Stonecutter#10]
When accessing inventory, minecraft crashes. below is the error.

Playing FTB Ocean block. I have repeatable on 2 versions 1.0 and 1.4 same bug. 

Will post whole debug log also.

```
24Oct2021 16:08:08.365] [Render thread/ERROR] [mezz.jei.load.PluginCaller/]: Caught an error from mod plugin: class xyz.brckts.portablestonecutter.compat.jei.JEIAddon portable_stonecutter:jei
java.lang.IllegalArgumentException: A RecipeCategory with UID "portable_stonecutter:anvil_flattening" has already been registered.
	at mezz.jei.load.registration.RecipeCategoryRegistration.addRecipeCategories(RecipeCategoryRegistration.java:38) ~[jei:7.7.1.126]
	at xyz.brckts.portablestonecutter.compat.jei.JEIAddon.registerCategories(JEIAddon.java:40) ~[portable_stonecutter:1.16.4-1.3.4]
	at mezz.jei.load.PluginLoader.lambda$new$3(PluginLoader.java:108) ~[jei:7.7.1.126]
	at mezz.jei.load.PluginCaller.callOnPlugins(PluginCaller.java:26) ~[jei:7.7.1.126]
	at mezz.jei.load.PluginLoader.<init>(PluginLoader.java:108) ~[jei:7.7.1.126]
	at mezz.jei.startup.JeiStarter.start(JeiStarter.java:69) ~[jei:7.7.1.126]
	at mezz.jei.startup.ClientLifecycleHandler.startJEI(ClientLifecycleHandler.java:147) ~[jei:7.7.1.126]
	at mezz.jei.startup.ClientLifecycleHandler.setupJEI(ClientLifecycleHandler.java:140) ~[jei:7.7.1.126]
	at mezz.jei.startup.ClientLifecycleHandler.lambda$new$3(ClientLifecycleHandler.java:106) ~[jei:7.7.1.126]
	at net.minecraftforge.eventbus.EventBus.doCastFilter(EventBus.java:247) ~[eventbus-4.0.0.jar:?]
	at net.minecraftforge.eventbus.EventBus.lambda$addListener$11(EventBus.java:239) ~[eventbus-4.0.0.jar:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-4.0.0.jar:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-4.0.0.jar:?]
	at net.minecraftforge.client.ForgeHooksClient.onRecipesUpdated(ForgeHooksClient.java:711) ~[forge:?]
	at net.minecraft.client.network.play.ClientPlayNetHandler.func_199525_a(ClientPlayNetHandler.java:1314) ~[?:?]
	at net.minecraft.network.play.server.SUpdateRecipesPacket.func_148833_a(SourceFile:27) ~[?:?]
	at net.minecraft.network.play.server.SUpdateRecipesPacket.func_148833_a(SourceFile:15) ~[?:?]
	at net.minecraft.network.PacketThreadUtil.lambda$checkThreadAndEnqueue$0(PacketThreadUtil.java:36) ~[?:?]
	at net.optifine.util.PacketRunnable.run(PacketRunnable.java:28) ~[?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213166_h(ThreadTaskExecutor.java:189) ~[?:?]
	at net.minecraft.util.concurrent.RecursiveEventLoop.func_213166_h(SourceFile:23) ~[?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213168_p(ThreadTaskExecutor.java:151) ~[?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213160_bf(ThreadTaskExecutor.java:128) ~[?:?]
	at net.minecraft.client.Minecraft.func_195542_b(Minecraft.java:947) ~[?:?]
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:607) ~[?:?]
	at net.minecraft.client.main.Main.main(Main.java:184) ~[1.16.5.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_242]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_242]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_242]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_242]
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider.lambda$launchService$0(FMLClientLaunchProvider.java:51) ~[forge-1.16.5-36.2.8.jar:36.2]
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-8.0.9.jar:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54) [modlauncher-8.0.9.jar:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72) [modlauncher-8.0.9.jar:?]
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:82) [modlauncher-8.0.9.jar:?]
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:66) [modlauncher-8.0.9.jar:?]
``
[debug.log](https://github.com/brckts/Portable-Stonecutter/files/7404160/debug.log)
`

**Comment by brckts** [comment]
Hello, thanks for the report. I have tried to reproduce your issue in oceanblock 1.4 and 1.5 but to no avail. Does this happen every time? Was it in singleplayer or on a server?

**Comment by Svennixx** [comment]
This also happens to me, playing on a server and using 1.41

**Comment by brckts** [comment]
Can you check if it happens for you on 1.5? I've just tried it on a server and couldn't replicate it.

**Comment by brckts** [comment]
> This also happens to me, playing on a server and using 1.41

I've just tried it on a 1.41 server and couldn't replicate it either... Is this on a fresh instance or did you update in place? The only thing I can think of that would cause this log is the mod being loaded twice.

**Comment by Svennixx** [comment]
> > This also happens to me, playing on a server and using 1.41
> 
> I've just tried it on a 1.41 server and couldn't replicate it either... Is this on a fresh instance or did you update in place? The only thing I can think of that would cause this log is the mod being loaded twice.

I just managed to fix it on my end, I was using adoptjava 15 and went back to java 8 and that solved it.

**Comment by brckts** [comment]
> > > This also happens to me, playing on a server and using 1.41
> > 
> > 
> > I've just tried it on a 1.41 server and couldn't replicate it either... Is this on a fresh instance or did you update in place? The only thing I can think of that would cause this log is the mod being loaded twice.
> 
> I just managed to fix it on my end, I was using adoptjava 15 and went back to java 8 and that solved it.

Glad to hear it!

**Comment by Raidobw2** [comment]
Hello. We have a user on version 1.5.0 of OceanBlock that got this issue too. All the info is in this link above ^.
Same stacktrace as the original one posted here.
They run Java 8, don't have any duplicate mod files, don't have Optifine.
I honestly have no idea if this is your mod doing this, but I've tried and gathered as much info as I could on that issue above in case that's helpful.
Let me know if you think it's something else, I might fill another issue to JEI anyways.
Thanks and have a nice day!

**Comment by brckts** [comment]
> Hello. We have a user on version 1.5.0 of OceanBlock that got this issue too. All the info is in this link above ^. Same stacktrace as the original one posted here. They run Java 8, don't have any duplicate mod files, don't have Optifine. I honestly have no idea if this is your mod doing this, but I've tried and gathered as much info as I could on that issue above in case that's helpful. Let me know if you think it's something else, I might fill another issue to JEI anyways. Thanks and have a nice day!

Thanks for the report. I'll investigate some more this evening then. Does this happen to everyone? Does this happen consistently for those affected? If only a specific subset of users is consistently affected it might be worth looking into what they did differently.

**Comment by brckts** [comment]
I think it has to do with AnvilFlatteningRecipeCategory and IAnvilFlatteningRecipe having the same UID, so I differentiated them to be sure. As I wasn't able to reproduce the issue I cannot be sure that it fixed it so if someone who had the issue can confirm for me I'd be grateful :+1:

**Comment by AlexandrXamlo** [comment]
> I think it has to do with AnvilFlatteningRecipeCategory and IAnvilFlatteningRecipe having the same UID, so I differentiated them to be sure. As I wasn't able to reproduce the issue I cannot be sure that it fixed it so if someone who had the issue can confirm for me I'd be grateful 👍

What should I do to help you?) I stopped playing OceanBlock because I caught crash every time.

**Comment by brckts** [comment]
> > I think it has to do with AnvilFlatteningRecipeCategory and IAnvilFlatteningRecipe having the same UID, so I differentiated them to be sure. As I wasn't able to reproduce the issue I cannot be sure that it fixed it so if someone who had the issue can confirm for me I'd be grateful +1
> 
> What should I do to help you?) I stopped playing OceanBlock because I caught crash every time.

Replace the Portable stonecutter jar file in your instance with the one [here](https://github.com/brckts/Portable-Stonecutter/releases/tag/1.3.5) and see if it works.

**Comment by brckts** [comment]
If it fixes it I'll release it on curseforge tonight so the pack devs can add it to the next release, if not I'll dig deeper.

**Comment by AlexandrXamlo** [comment]
> If it fixes it I'll release it on curseforge tonight so the pack devs can add it to the next release, if not I'll dig deeper.

Well. Give me some time to see how the game will behave with this version

**Comment by AlexandrXamlo** [comment]
Crash again(. Here is my latest log
[latest.log](https://github.com/brckts/Portable-Stonecutter/files/7444680/latest.log)

**Comment by brckts** [comment]
> Crash again(. Here is my latest log [latest.log](https://github.com/brckts/Portable-Stonecutter/files/7444680/latest.log)

Thanks for taking the time to try it. I looked at your log and it doesn't indicate a crash. Do you have a crashlog somewhere?

**Comment by AlexandrXamlo** [comment]
> > Crash again(. Here is my latest log [latest.log](https://github.com/brckts/Portable-Stonecutter/files/7444680/latest.log)
> 
> Thanks for taking the time to try it. I looked at your log and it doesn't indicate a crash. Do you have a crashlog somewhere?

I also noticed that the log is different from what I uploaded earlier (in another topic). But I don't find any other. There is also a debug log, but there is nothing in it either.
In the last log there is a line where ERROR is written in quotes. Could it have caused the fall?

**Comment by AlexandrXamlo** [comment]
> > Crash again(. Here is my latest log [latest.log](https://github.com/brckts/Portable-Stonecutter/files/7444680/latest.log)
> 
> Thanks for taking the time to try it. I looked at your log and it doesn't indicate a crash. Do you have a crashlog somewhere?

in the root with the modpack I found such a file - hs_err_pid16380.log 
Can it help in any way?
[hs_err_pid16380.log](https://github.com/brckts/Portable-Stonecutter/files/7444754/hs_err_pid16380.log)

**Comment by brckts** [comment]
> > > Crash again(. Here is my latest log [latest.log](https://github.com/brckts/Portable-Stonecutter/files/7444680/latest.log)
> > 
> > 
> > Thanks for taking the time to try it. I looked at your log and it doesn't indicate a crash. Do you have a crashlog somewhere?
> 
> in the root with the modpack I found such a file - hs_err_pid16380.log Can it help in any way? [hs_err_pid16380.log](https://github.com/brckts/Portable-Stonecutter/files/7444754/hs_err_pid16380.log)

that looks like a jvm crash log but I have no idea how or if I can glean information from those....

I looked a bit deeper at the debug log from the OP, it somehow registers item subtypes, fluid subtypes, ingredients and categories twice, but vanilla category extensions, recipes, recipe transfer handlers, recipe catalysts, gui handlers and advanced plugins only once :thinking: 

It makes no sense to me as to why it would load the first ones twice (thus crashing jei) but the second ones only once...

The other thread mentions JEIIntegration as a possible culprit, does it still crash if you disable it?

**Comment by AlexandrXamlo** [comment]
> > > > Crash again(. Here is my latest log [latest.log](https://github.com/brckts/Portable-Stonecutter/files/7444680/latest.log)
> > > 
> > > 
> > > Thanks for taking the time to try it. I looked at your log and it doesn't indicate a crash. Do you have a crashlog somewhere?
> > 
> > 
> > in the root with the modpack I found such a file - hs_err_pid16380.log Can it help in any way? [hs_err_pid16380.log](https://github.com/brckts/Portable-Stonecutter/files/7444754/hs_err_pid16380.log)
> 
> that looks like a jvm crash log but I have no idea how or if I can glean information from those....
> 
> I looked a bit deeper at the debug log from the OP, it somehow registers item subtypes, fluid subtypes, ingredients and categories twice, but vanilla category extensions, recipes, recipe transfer handlers, recipe catalysts, gui handlers and advanced plugins only once 🤔
> 
> It makes no sense to me as to why it would load the first ones twice (thus crashing jei) but the second ones only once...
> 
> The other thread mentions JEIIntegration as a possible culprit, does it still crash if you disable it?

I deleted JEIIntegration from the mods folder, enter the world, open crafting grid, type "6" in search and crash...

**Comment by brckts** [comment]
> > > > > Crash again(. Here is my latest log [latest.log](https://github.com/brckts/Portable-Stonecutter/files/7444680/latest.log)
> > > > 
> > > > 
> > > > Thanks for taking the time to try it. I looked at your log and it doesn't indicate a crash. Do you have a crashlog somewhere?
> > > 
> > > 
> > > in the root with the modpack I found such a file - hs_err_pid16380.log Can it help in any way? [hs_err_pid16380.log](https://github.com/brckts/Portable-Stonecutter/files/7444754/hs_err_pid16380.log)
> > 
> > 
> > that looks like a jvm crash log but I have no idea how or if I can glean information from those....
> > I looked a bit deeper at the debug log from the OP, it somehow registers item subtypes, fluid subtypes, ingredients and categories twice, but vanilla category extensions, recipes, recipe transfer handlers, recipe catalysts, gui handlers and advanced plugins only once thinking
> > It makes no sense to me as to why it would load the first ones twice (thus crashing jei) but the second ones only once...
> > The other thread mentions JEIIntegration as a possible culprit, does it still crash if you disable it?
> 
> I deleted JEIIntegration from the mods folder, enter the world, open crafting grid, type "6" in search and crash...

Does it only happen in the RS crafting grid and when searching for 6? Is your crafting grid on JEI Synchronised mode?

**Comment by AlexandrXamlo** [comment]
Yes, RF crafting grid is synchronized with JEI (two-way). But you know, I opened the world, scrolled through all the pages of JEI, type "6" in the search, opened crafting grid, entered "6" and nothing happens. The game doesn't crash.

> > > > > > Crash again(. Here is my latest log [latest.log](https://github.com/brckts/Portable-Stonecutter/files/7444680/latest.log)
> > > > > 
> > > > > 
> > > > > Thanks for taking the time to try it. I looked at your log and it doesn't indicate a crash. Do you have a crashlog somewhere?
> > > > 
> > > > 
> > > > in the root with the modpack I found such a file - hs_err_pid16380.log Can it help in any way? [hs_err_pid16380.log](https://github.com/brckts/Portable-Stonecutter/files/7444754/hs_err_pid16380.log)
> > > 
> > > 
> > > that looks like a jvm crash log but I have no idea how or if I can glean information from those....
> > > I looked a bit deeper at the debug log from the OP, it somehow registers item subtypes, fluid subtypes, ingredients and categories twice, but vanilla category extensions, recipes, recipe transfer handlers, recipe catalysts, gui handlers and advanced plugins only once thinking
> > > It makes no sense to me as to why it would load the first ones twice (thus crashing jei) but the second ones only once...
> > > The other thread mentions JEIIntegration as a possible culprit, does it still crash if you disable it?
> > 
> > 
> > I deleted JEIIntegration from the mods folder, enter the world, open crafting grid, type "6" in search and crash...
> 
> Does it only happen in the RS crafting grid and when searching for 6? Is your crafting grid on JEI Synchronised mode?

Yes, RS crafting grid is synchronized with JEI (two-way). 
But you know, I opened the world, scrolled through all the pages of JAY, type "6" in the search, opened crafting grid, entered "6" and nothing happens. The game doesn't crash.
Now I will try to turn JEIIntegration back on...

**Comment by AlexandrXamlo** [comment]
If I scroll through the entire JEI after entering the world, then no error appears...

**Comment by Raidobw2** [comment]
@AlexandrXamlo I think the version the author gave you might have solved it. There's another bug at FTBTeam/FTB-Modpack-Issues#40, and it's the reason for your Java crash earlier today. :S

Fix: go in `config/brandon3055/DraconicEvolution.cfg` and disable all those (they are set to "true", set them all to "false"):
```
# Set this to false to disable tool shaders.
B:"toolShaders"=true

# Set this to false to disable crystal shaders.
B:"crystalShaders"=true

# Set this to false to disable reactor shaders.
B:"reactorShaders"=true

# Set this to false to disable all other shaders.
B:"otherShaders"=true
```
-----------------
Sorry about that, once that's done you should be able to start your game and type in your Refined Storage again and report to the author here if their dev version worked! :)

**Comment by AlexandrXamlo** [comment]
> @AlexandrXamlo I think the version the author gave you might have solved it. There's another bug at [FTBTeam/FTB-Modpack-Issues#40](https://github.com/FTBTeam/FTB-Modpack-Issues/issues/40), and it's the reason for your Java crash earlier today. :S
> 
> Fix: go in `config/brandon3055/DraconicEvolution.cfg` and disable all those (they are set to "true", set them all to "false"):
> 
> ```
> # Set this to false to disable tool shaders.
> B:"toolShaders"=true
> 
> # Set this to false to disable crystal shaders.
> B:"crystalShaders"=true
> 
> # Set this to false to disable reactor shaders.
> B:"reactorShaders"=true
> 
> # Set this to false to disable all other shaders.
> B:"otherShaders"=true
> ```
> 
> Sorry about that, once that's done you should be able to start your game and type in your Refined Storage again and report to the author here if their dev version worked! :)

It helped! Thank you for your promptness and help in solving the problem.

**Comment by brckts** [comment]
Thank you @AlexandrXamlo for helping test things, I guess we can close this issue now and release 1.3.5
